packages/components/src/unit-control/types.ts

```diff
@@ -24,3 +37,43 @@ export type WPUnitControlUnit = {
 };
 
 export type WPUnitControlUnitList = Array< WPUnitControlUnit > | false;
+
+export interface UnitControlOnChangeCallback {
+	(
+		nextValue: string | undefined,
+		extra: {
+			event: SyntheticEvent< HTMLSelectElement | HTMLInputElement >;
+			data?: WPUnitControlUnit;
+		}
+	): void;
+}
+
+export type UnitSelectControlProps = {
+	className?: string;
+	disabled?: boolean;
+	isTabbable: boolean;
+	onChange?: UnitControlOnChangeCallback;
+	options?: WPUnitControlUnitList;
+	size?: SelectSize;
+	value?: string;
+};
+
+export interface UnitControlProps {
```

Comment:
I was comparing this component to the InputControl component and went with `interface` for consistency, but given that we don't need to extend or inherit, perhaps this one can be a `type`, too?

Reply:
How much overlap is there between `InputControl`'s types and `UnitControl`'s types? It would be interesting to potentially re-use some of `InputControl` types in `UnitControl`.

On a similar note, I looked at `UnitSelectControlProps` and `UnitControlProps` and realised that, if we rename a couple of props (`isTabbable` => `isUnitSelectTabbable`, `options` => `units`, and `value` => `unit`), we can simplify these 2 types and make `UnitControlProps` extend `UnitSelectControlProps`:

<details>
<summary>Click to expand</summary>

```diff
diff --git a/packages/components/src/unit-control/index.tsx b/packages/components/src/unit-control/index.tsx
index 7684ccfc28..fa09e4c210 100644
--- a/packages/components/src/unit-control/index.tsx
+++ b/packages/components/src/unit-control/index.tsx
@@ -118,7 +118,7 @@ function UnitControl(
 			unit
 		);
 
-		refParsedValue.current = parsedValue?.toString();
+		refParsedValue.current = parsedValue.toString();
 
 		if ( isPressEnterToChange && parsedUnit !== unit ) {
 			const data = Array.isArray( units )
@@ -171,11 +171,11 @@ function UnitControl(
 		<UnitSelectControl
 			aria-label={ __( 'Select unit' ) }
 			disabled={ disabled }
-			isTabbable={ isUnitSelectTabbable }
-			options={ units }
+			isUnitSelectTabbable={ isUnitSelectTabbable }
+			units={ units }
 			onChange={ handleOnUnitChange }
 			size={ size }
-			value={ unit }
+			unit={ unit }
 		/>
 	) : null;
 
diff --git a/packages/components/src/unit-control/types.ts b/packages/components/src/unit-control/types.ts
index d843266fc9..768fa92213 100644
--- a/packages/components/src/unit-control/types.ts
+++ b/packages/components/src/unit-control/types.ts
@@ -51,29 +51,22 @@ export interface UnitControlOnChangeCallback {
 export type UnitSelectControlProps = {
 	className?: string;
 	disabled?: boolean;
-	isTabbable: boolean;
+	isUnitSelectTabbable: boolean;
 	onChange?: UnitControlOnChangeCallback;
-	options?: WPUnitControlUnitList;
 	size?: SelectSize;
-	value?: string;
+	units?: WPUnitControlUnitList;
+	unit?: string;
 };
 
-export interface UnitControlProps {
+export type UnitControlProps = UnitSelectControlProps & {
 	__unstableStateReducer?: StateReducer;
 	autoComplete?: string;
-	className?: string;
-	disabled?: boolean;
 	disableUnits?: boolean;
 	isPressEnterToChange?: boolean;
 	isResetValueOnUnitChange?: boolean;
-	isUnitSelectTabbable?: boolean;
 	label?: string;
-	onChange?: UnitControlOnChangeCallback;
 	onUnitChange?: UnitControlOnChangeCallback;
-	size?: SelectSize;
 	step?: number;
 	style?: CSSProperties;
-	unit?: string;
-	units?: WPUnitControlUnitList;
 	value: string;
-}
+};
diff --git a/packages/components/src/unit-control/unit-select-control.tsx b/packages/components/src/unit-control/unit-select-control.tsx
index ea6d9554a3..3f40edcb9d 100644
--- a/packages/components/src/unit-control/unit-select-control.tsx
+++ b/packages/components/src/unit-control/unit-select-control.tsx
@@ -17,37 +17,37 @@ import type { UnitSelectControlProps } from './types';
  * Renders a `select` if there are multiple units.
  * Otherwise, renders a non-selectable label.
  *
- * @param  props            Component props.
- * @param  props.className  Class to set on the `select` element.
- * @param  props.isTabbable Whether the control can be focused via keyboard navigation.
- * @param  props.options    Available units to select from.
- * @param  props.onChange   A callback function invoked when the value is changed.
- * @param  props.size       Size of the control option. Supports "default" and "small".
- * @param  props.value      Current unit.
+ * @param  props                      Component props.
+ * @param  props.className            Class to set on the `select` element.
+ * @param  props.isUnitSelectTabbable Whether the control can be focused via keyboard navigation.
+ * @param  props.onChange             A callback function invoked when the value is changed.
+ * @param  props.size                 Size of the control option. Supports "default" and "small".
+ * @param  props.unit                 Current unit.
+ * @param  props.units                Available units to select from.
  */
 export default function UnitSelectControl( {
 	className,
-	isTabbable = true,
-	options = CSS_UNITS,
+	isUnitSelectTabbable: isTabbable = true,
 	onChange = noop,
 	size = 'default',
-	value = 'px',
+	unit = 'px',
+	units = CSS_UNITS,
 	...props
 }: UnitSelectControlProps ) {
-	if ( ! options || ! hasUnits( options ) || options?.length === 1 ) {
+	if ( ! units || ! hasUnits( units ) || units?.length === 1 ) {
 		return (
 			<UnitLabel
 				className="components-unit-control__unit-label"
 				selectSize={ size }
 			>
-				{ value }
+				{ unit }
 			</UnitLabel>
 		);
 	}
 
 	const handleOnChange = ( event: ChangeEvent< HTMLSelectElement > ) => {
 		const { value: unitValue } = event.target;
-		const data = options.find( ( option ) => option.value === unitValue );
+		const data = units.find( ( option ) => option.value === unitValue );
 
 		onChange( unitValue, { event, data } );
 	};
@@ -60,10 +60,10 @@ export default function UnitSelectControl( {
 			onChange={ handleOnChange }
 			selectSize={ size }
 			tabIndex={ isTabbable ? undefined : -1 }
-			value={ value }
+			value={ unit }
 			{ ...props }
 		>
-			{ options.map( ( option ) => (
+			{ units.map( ( option ) => (
 				<option value={ option.value } key={ option.value }>
 					{ option.label }
 				</option>

```

</details>